kroxylicious-filter-test-support/src/main/java/io/kroxylicious/test/record/RecordTestUtils.java

```diff
@@ -295,7 +308,54 @@ public static MemoryRecords memoryRecords(byte magic, long offset, long timestam
     }
 
     private static MemoryRecords memoryRecordsWithoutCopy(byte magic, long offset, long timestamp, byte[] key, byte[] value, Header... headers) {
-        try (MemoryRecordsBuilder memoryRecordsBuilder = new MemoryRecordsBuilder(
+        try (MemoryRecordsBuilder memoryRecordsBuilder = defaultMemoryRecordsBuilder(magic)) {
+            memoryRecordsBuilder.appendWithOffset(offset, timestamp, key, value, headers);
+            return memoryRecordsBuilder.build();
+        }
+    }
+
+    /**
+     * Return a MemoryRecords containing a single RecordBatch containing multiple Records.
+     * The batch will use the current magic.
+     * @param records
+     * @return The MemoryRecords
+     */
+    public static MemoryRecords memoryRecords(Collection<Record> records) {
```

Comment:
I would expect that records should be ordered (i.e. List rather than Collection).